pkg/openapi/server.spec.yaml

```diff
@@ -159,8 +159,97 @@ components:
           description: Flavor ID.
           type: string
           minLength: 1
+        firewall:
+          description: A list of firewall rules applied to a workload pool.
+          type: object
+          properties:
+            ingress:
+              $ref: '#/components/schemas/firewallRules'
+        publicIPAllocation:
+          $ref: '#/components/schemas/publicIPAllocation'
+        image:
+          $ref: '#/components/schemas/imageSelector'
+        ssh:
+          description: SSH settings.
+          type: object
+          properties:
+            publicKeys:
+              description: A list of public SSH keys to allow access to the machine.
+              type: array
+              items:
+                type: string
+    firewallRules:
+      description: A list of firewall rules applied to a workload pool.
+      type: array
+      items:
+        $ref: '#/components/schemas/firewallRule'
+    firewallRule:
+      description: A firewall rule applied to a workload pool.
+      type: object
+      required:
+      - protocol
+      - port
+      - cidr
+      properties:
+        protocol:
+          description: The protocol to allow.
+          type: string
+          enum:
+          - tcp
+          - udp
+        port:
+          $ref: '#/components/schemas/firewallRulePort'
+        cidr:
+          description: A list of CIDR blocks to allow, it might be any IPv4 or IPv6 in CIDR notation.
```

Comment:
Surprise I've not done this yet, but we should probably have a type definition in core for an IPv4 CIDR with the required regular expression patter matcher too.